README.md
# :globe_with_meridians: http

[![Packagist License](https://img.shields.io/packagist/l/folded/http)](https://github.com/folded-php/http/blob/next/LICENSE) ![Packagist PHP Version Support](https://img.shields.io/packagist/php-v/folded/http) [![Packagist Version](https://img.shields.io/packagist/v/folded/http)](https://packagist.org/packages/folded/http) [![Unit Tests](https://github.com/folded-php/http/workflows/Unit%20Tests/badge.svg)](https://github.com/folded-php/http/actions?query=workflow%3A%22Unit+Tests%22) [![Maintainability](https://api.codeclimate.com/v1/badges/eded14cbb9d6cc3eec45/maintainability)](https://codeclimate.com/github/folded-php/http/maintainability)

HTTP utilities for your web app.

## Summary

- [Installation](#installation)
- [Requirements](#requirements)
- [API](#api)
  - [Information response](#information-response)
    - [Continue](#continue)
    - [Switching protocols](#switching-protocols)
    - [Processing](#processing)
    - [Early hints](#early-hints)
  - [Success](#success)
    - [Created](#created)
    - [Accepted](#accepted)
    - [Non-authoritative information](#non-authoritative-information)
    - [No content](#no-content)
    - [Reset content](#reset-content)
    - [Partial content](#partial-content)
    - [Multi-status](#multi-status)
    - [Already reported](#already-reported)
    - [Instance-manipulations used](#instance-manipulations-used)
  - [Redirection](#redirection)
    - [Multiple choices](#multiple-choices)
    - [Moved permanently](#moved-permanently)
    - [Found (moved temporarily)](#found-moved-temporarily)
    - [See other](#see-other)
    - [Not modified](#not-modified)
    - [Use proxy](#use-proxy)
    - [Switch proxy](#switch-proxy)
    - [Temporary redirect](#temporary-redirect)
    - [Permanent redirect](#permanent-redirect)
  - [Error](#error)
    - [Bad request](#bad-request)
    - [Unauthorized](#unauthorized)
    - [Payment required](#payment-required)
    - [Forbidden](#forbidden)
    - [Not found](#not-found)
    - [Method not allowed](#method-not-allowed)
    - [Not acceptable](#not-acceptable)
    - [Proxy authentication required](#proxy-authentication-required)
    - [Request timeout](#request-timeout)
    - [Conflict](#conflict)
    - [Gone](#gone)
    - [Length required](#length-required)
    - [Precondition failed](#precondition-failed)
    - [Payload too large](#payload-too-large)
    - [URI too long](#uri-too-long)
    - [Unsupported media type](#unsupported-media-type)
    - [Range not satisfiable](#range-not-satisfiable)
    - [Expectation failed](#expectation-failed)
    - [I'm a teapot](#im-a-teapot)
    - [Misdirected request](#misdirected-request)
    - [Unprocessable entity](#unprocessable-entity)
    - [Locked](#locked)
    - [Failed dependency](#failed-dependency)
    - [Too early](#too-early)
    - [Upgrade required](#upgrade-required)
    - [Precondition required](#precondition-failed)
    - [Too many requests](#too-many-requests)
    - [Request header fields too large](#request-header-fields-too-large)
    - [Unavailable for legal reasons](#unavailable-for-legal-reasons)
  - [Server error](#server-error)
    - [Internal server error](#internal-server-error)
    - [Not implemented](#not-implemented)
    - [Bad gateway](#bad-gateway)
    - [Service unavailable](#service-unavailable)
    - [Gateway timeout](#gateway-timeout)
    - [Http version not supported](#http-version-not-supported)
    - [Variant also negotiates](#variant-also-negotiates)
    - [Insufficient storage](#insufficient-storage)
    - [Loop detected](#loop-detected)
    - [Not extended](#not-extended)
    - [Network authentication required](#network-authentication-required)
- [Changelog](#changelog)
- [Contributing](#contributing)
- [License](#license)

## Requirements

- [PHP 7.1 (at least)](https://www.php.net/downloads)
- [Composer](https://getcomposer.org/)

## Installation

```console
$ composer require folded/http
```

[:arrow_up: Back to summary](#summary).

## API

[:arrow_up: Back to summary](#summary).

### Information response

[:arrow_up: Back to summary](#summary).

#### Continue

```php
<?php

use Http\Information;

echo Information::CONTINUE; // 100
```

[:arrow_up: Back to summary](#summary).

#### Switching protocols

```php
<?php

use Http\Information;

echo Information::SWITCHING_PROTOCOLS; // 101
```

[:arrow_up: Back to summary](#summary).

#### Processing

```php
<?php

use Http\Information;

echo Information::PROCESSING; // 102
```

[:arrow_up: Back to summary](#summary).

#### Early hints

```php
<?php

use Http\Information;

echo Information::EARLY_HINTS; // 103
```

[:arrow_up: Back to summary](#summary).

### Success

```php
<?php

use Http\Success;

echo Success::OK; // 200
```

[:arrow_up: Back to summary](#summary).

#### Created

```php
<?php

use Http\Success;

echo Success::CREATED; // 201
```

[:arrow_up: Back to summary](#summary).

#### Accepted

```php
<?php

use Http\Success;

echo Success::ACCEPTED; // 202
```

[:arrow_up: Back to summary](#summary).

#### Non-authoritative information

```php
<?php

use Http\Success;

echo Success::NON_AUTHORITATIVE_INFORMATION; // 203
```

[:arrow_up: Back to summary](#summary).

#### No content

```php
<?php

use Http\Success;

echo Success::NO_CONTENT; // 204
```

[:arrow_up: Back to summary](#summary).

#### Reset content

```php
<?php

use Http\Success;

echo Success::RESET_CONTENT; // 205
```

[:arrow_up: Back to summary](#summary).

#### Partial content

```php
<?php

use Http\Success;

echo Success::PARTIAL_CONTENT; // 206
```

[:arrow_up: Back to summary](#summary).

#### Multi-status

```php
<?php

use Http\Success;

echo Success::MULTI_STATUS; // 207
```

[:arrow_up: Back to summary](#summary).

#### Already reported

```php
<?php

use Http\Success;

echo Success::ALREADY_REPORTED; // 208
```

[:arrow_up: Back to summary](#summary).

#### Instance-manipulations used

```php
<?php

use Http\Success;

echo Success::IM_USED; // 226
```

[:arrow_up: Back to summary](#summary).

### Redirection

[:arrow_up: Back to summary](#summary).

#### Multiple choices

```php
<?php

use Http\Redirection;

echo Redirection::MULTIPLE_CHOICES; // 300
```

[:arrow_up: Back to summary](#summary).

#### Moved permanently

```php
<?php

use Http\Redirection;

echo Redirection::MOVED_PERMANENTLY; // 301
```

[:arrow_up: Back to summary](#summary).

#### Found (moved temporarily)

```php
<?php

use Http\Redirection;

echo Redirection::FOUND;                // 302
echo Redirection::MOVED_TEMPORARILY;    // 302
```

[:arrow_up: Back to summary](#summary).

#### See other

```php
<?php

use Http\Redirection;

echo Redirection::SEE_OTHER; // 303
```

[:arrow_up: Back to summary](#summary).

#### Not modified

```php
<?php

use Http\Redirection;

echo Redirection::NOT_MODIFIED; // 304
```

[:arrow_up: Back to summary](#summary).

#### Use proxy

```php
<?php

use Http\Redirection;

echo Redirection::USE_PROXY; // 305
```

[:arrow_up: Back to summary](#summary).

#### Switch proxy

```php
<?php

use Http\Redirection;

echo Redirection::SWITCH_PROXY; // 306
```

[:arrow_up: Back to summary](#summary).

#### Temporary redirect

```php
<?php

use Http\Redirection;

echo Redirection::TEMPORARY_REDIRECT; // 307
```

[:arrow_up: Back to summary](#summary).

#### Permanent redirect

```php
<?php

use Http\Redirection;

echo Redirection::PERMANENT_REDIRECT; // 308
```

[:arrow_up: Back to summary](#summary).

### Error

[:arrow_up: Back to summary](#summary).

#### Bad request

```php
<?php

use Http\Error;

echo Error::BAD_REQUEST; // 400
```

[:arrow_up: Back to summary](#summary).

#### Unauthorized

```php
<?php

use Http\Error;

echo Error::UNAUTHORIZED; // 401
```

[:arrow_up: Back to summary](#summary).

#### Payment required

```php
<?php

use Http\Error;

echo Error::PAYMENT_REQUIRED; // 402
```

[:arrow_up: Back to summary](#summary).

#### Forbidden

```php
<?php

use Http\Error;

echo Error::FORBIDDEN; // 403
```

[:arrow_up: Back to summary](#summary).

#### Not found

```php
<?php

use Http\Error;

echo Error::NOT_FOUND; // 404
```

[:arrow_up: Back to summary](#summary).

#### Method not allowed

```php
<?php

use Http\Error;

echo Error::METHOD_NOT_ALLOWED; // 405
```

[:arrow_up: Back to summary](#summary).

#### Not acceptable

```php
<?php

use Http\Error;

echo Error::NOT_ACCEPTABLE; // 406
```

[:arrow_up: Back to summary](#summary).

#### Proxy authentication required

```php
<?php

use Http\Error;

echo Error::PROXY_AUTHENTICATION_REQUIRED; // 407
```

[:arrow_up: Back to summary](#summary).

#### Request timeout

```php
<?php

use Http\Error;

echo Error::REQUEST_TIMEOUT; // 408
```

[:arrow_up: Back to summary](#summary).

#### Conflict

```php
<?php

use Http\Error;

echo Error::CONFLICT; // 409
```

[:arrow_up: Back to summary](#summary).

#### Gone

```php
<?php

use Http\Error;

echo Error::GONE; // 410
```

[:arrow_up: Back to summary](#summary).

#### Length required

```php
<?php

use Http\Error;

echo Error::LENGTH_REQUIRED; // 411
```

[:arrow_up: Back to summary](#summary).

#### Precondition failed

```php
<?php

use Http\Error;

echo Error::PRECONDITION_FAILED; // 412
```

[:arrow_up: Back to summary](#summary).

#### Payload too large

```php
<?php

use Http\Error;

echo Error::PAYLOAD_TOO_LARGE; // 413
```

[:arrow_up: Back to summary](#summary).

#### URI too long

```php
<?php

use Http\Error;

echo Error::URI_TOO_LONG; // 414
```

[:arrow_up: Back to summary](#summary).

#### Unsupported media type

```php
<?php

use Http\Error;

echo Error::UNSUPPORTED_MEDIA_TYPE; // 415
```

[:arrow_up: Back to summary](#summary).

#### Range not satisfiable

```php
<?php

use Http\Error;

echo Error::RANGE_NOT_SATISFIABLE; // 416
```

[:arrow_up: Back to summary](#summary).

#### Expectation failed

```php
<?php

use Http\Error;

echo Error::EXPECTATION_FAILED; // 417
```

[:arrow_up: Back to summary](#summary).

#### I'm a teapot

```php
<?php

use Http\Error;

echo Error::IM_A_TEAPOT; // 418
```

[:arrow_up: Back to summary](#summary).

#### Misdirected request

```php
<?php

use Http\Error;

echo Error::MISDIRECTED_REQUEST; // 419
```

[:arrow_up: Back to summary](#summary).

#### Unprocessable entity

```php
<?php

use Http\Error;

echo Error::UNPROCESSABLE_ENTITY; // 422
```

[:arrow_up: Back to summary](#summary).

#### Locked

```php
<?php

use Http\Error;

echo Error::LOCKED; // 423
```

[:arrow_up: Back to summary](#summary).

#### Failed dependency

```php
<?php

use Http\Error;

echo Error::FAILED_DEPENDENCY; // 424
```

[:arrow_up: Back to summary](#summary).

#### Too early

```php
<?php

use Http\Error;

echo Error::TOO_EARLY; // 425
```

[:arrow_up: Back to summary](#summary).

#### Upgrade required

```php
<?php

use Http\Error;

echo Error::UPGRADE_REQUIRED; // 426
```

[:arrow_up: Back to summary](#summary).

#### Precondition required

```php
<?php

use Http\Error;

echo Error::PRECONDITION_REQUIRED; // 428
```

[:arrow_up: Back to summary](#summary).

#### Too many requests

```php
<?php

use Http\Error;

echo Error::TOO_MANY_REQUESTS; // 429
```

[:arrow_up: Back to summary](#summary).

#### Request header fields too large

```php
<?php

use Http\Error;

echo Error::REQUEST_HEADER_FIELDS_TOO_LARGE; // 431
```

[:arrow_up: Back to summary](#summary).

#### Unavailable for legal reasons

```php
<?php

use Http\Error;

echo Error::UNAVAILABLE_FOR_LEGAL_REASONS; // 451
```

[:arrow_up: Back to summary](#summary).

### Server error

[:arrow_up: Back to summary](#summary).

#### Internal server error

```php
<?php

use Http\ServerError;

echo ServerError::INTERNAL_SERVER_ERROR; // 500
```

[:arrow_up: Back to summary](#summary).

#### Not implemented

```php
<?php

use Http\ServerError;

echo ServerError::NOT_IMPLEMENTED; // 501
```

[:arrow_up: Back to summary](#summary).

#### Bad gateway

```php
<?php

use Http\ServerError;

echo ServerError::BAD_GATEWAY; // 502
```

[:arrow_up: Back to summary](#summary).

#### Service unavailable

```php
<?php

use Http\ServerError;

echo ServerError::SERVICE_UNAVAILABLE; // 503
```

[:arrow_up: Back to summary](#summary).

#### Gateway timeout

```php
<?php

use Http\ServerError;

echo ServerError::GATEWAY_TIMEOUT; // 504
```

[:arrow_up: Back to summary](#summary).

#### Http version not supported

```php
<?php

use Http\ServerError;

echo ServerError::HTTP_VERSION_NOT_SUPPORTED; // 505
```

[:arrow_up: Back to summary](#summary).

#### Variant also negotiates

```php
<?php

use Http\ServerError;

echo ServerError::VARIANT_ALSO_NEGOTIATES; // 506
```

[:arrow_up: Back to summary](#summary).

#### Insufficient storage

```php
<?php

use Http\ServerError;

echo ServerError::INSUFFICIENT_STORAGE; // 507
```

[:arrow_up: Back to summary](#summary).

#### Loop detected

```php
<?php

use Http\ServerError;

echo ServerError::LOOP_DETECTED; // 508
```

[:arrow_up: Back to summary](#summary).

#### Not extended

```php
<?php

use Http\ServerError;

echo ServerError::NOT_EXTENDED; // 510
```

[:arrow_up: Back to summary](#summary).

#### Network authentication required

```php
<?php

use Http\ServerError;

echo ServerError::NETWORK_AUTHENTICATION_REQUIRED; // 511
```

[:arrow_up: Back to summary](#summary).

## Changelog

See [`CHANGELOG.md`](./CHANGELOG.md).

[:arrow_up: Back to summary](#summary).

## Contributing

See [`CONTRIBUTING.md`](./CONTRIBUTING.md).

[:arrow_up: Back to summary](#summary).

## License

See [`LICENSE`](./LICENSE).

[:arrow_up: Back to summary](#summary).


CHANGELOG.md
# Changelog

## Summary

- [`0.1.2`](#011)
- [`0.1.1`](#011)
- [`0.1.0`](#010)

## `0.1.2`

- :bug: fixed an issue with the homepage of the composer configuration ([#5](../../pull/5))
- :bug: fixed an issue with the issue link in the composer configuration ([#7](../../pull/7))
- :sparkles: fixed an issue with the source link in the composer configuration ([#6](../../pull/6))

## `0.1.1`

- Added php requirements in the Composer configuration.
- Added a license badge in the documentation.
- Added package version badge in the documentation.
- Added maintainability badge in the documentation.
- Added a command to remove all files listed in the `.gitignore` file.
- Added a service to issue shell commands in the Docker Compose configuration.
- Added a Dockerfile for the shell image used in the Docker Compose configuration.


## `0.1.0`

- Initial release.


tests/HttpErrorTest.php
<?php

/*!
 * Copyright © 2020 Amin NAIRI
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

use PHPUnit\Framework\TestCase;
use Folded\Http\Error;

final class HttpErrorTest extends TestCase
{
    public final function testBadRequest(): void
    {
        $this->assertEquals(Error::BAD_REQUEST, 400);
    }

    public final function testUnauthorized(): void
    {
        $this->assertEquals(Error::UNAUTHORIZED, 401);
    }

    public final function testPaymentRequired(): void
    {
        $this->assertEquals(Error::PAYMENT_REQUIRED, 402);
    }

    public final function testForbidden(): void
    {
        $this->assertEquals(Error::FORBIDDEN, 403);
    }

    public final function testNotFound(): void
    {
        $this->assertEquals(Error::NOT_FOUND, 404);
    }

    public final function testMethodNotAllowed(): void
    {
        $this->assertEquals(Error::METHOD_NOT_ALLOWED, 405);
    }

    public final function testNotAcceptable(): void
    {
        $this->assertEquals(Error::NOT_ACCEPTABLE, 406);
    }

    public final function testProxyAuthenticationRequired(): void
    {
        $this->assertEquals(Error::PROXY_AUTHENTICATION_REQUIRED, 407);
    }

    public final function testRequestTimeout(): void
    {
        $this->assertEquals(Error::REQUEST_TIMEOUT, 408);
    }

    public final function testConflict(): void
    {
        $this->assertEquals(Error::CONFLICT, 409);
    }

    public final function testGone(): void
    {
        $this->assertEquals(Error::GONE, 410);
    }

    public final function testLengthRequired(): void
    {
        $this->assertEquals(Error::LENGTH_REQUIRED, 411);
    }

    public final function testPreconditionFailed(): void
    {
        $this->assertEquals(Error::PRECONDITION_FAILED, 412);
    }

    public final function testPayloadTooLarge(): void
    {
        $this->assertEquals(Error::PAYLOAD_TOO_LARGE, 413);
    }

    public final function testUriTooLong(): void
    {
        $this->assertEquals(Error::URI_TOO_LONG, 414);
    }

    public final function testUnsupportedMediaType(): void
    {
        $this->assertEquals(Error::UNSUPPORTED_MEDIA_TYPE, 415);
    }

    public final function testRangeNotSatisfiable(): void
    {
        $this->assertEquals(Error::RANGE_NOT_SATISFIABLE, 416);
    }

    public final function testExpectationFailed(): void
    {
        $this->assertEquals(Error::EXPECTATION_FAILED, 417);
    }

    public final function testImATeaport(): void
    {
        $this->assertEquals(Error::IM_A_TEAPOT, 418);
    }

    public final function testMisdirectedRequest(): void
    {
        $this->assertEquals(Error::MISDIRECTED_REQUEST, 421);
    }

    public final function testUnprocessableEntity(): void
    {
        $this->assertEquals(Error::UNPROCESSABLE_ENTITY, 422);
    }

    public final function testLocked(): void
    {
        $this->assertEquals(Error::LOCKED, 423);
    }

    public final function testFailedDependency(): void
    {
        $this->assertEquals(Error::FAILED_DEPENDENCY, 424);
    }

    public final function testTooEearly(): void
    {
        $this->assertEquals(Error::TOO_EARLY, 425);
    }

    public final function testUpgradeRequired(): void
    {
        $this->assertEquals(Error::UPGRADE_REQUIRED, 426);
    }

    public final function testPreconditionRequired(): void
    {
        $this->assertEquals(Error::PRECONDITION_REQUIRED, 428);
    }

    public final function testTooManyRequests(): void
    {
        $this->assertEquals(Error::TOO_MANY_REQUESTS, 429);
    }

    public final function testRequestHeaderFieldsTooLarge(): void
    {
        $this->assertEquals(Error::REQUEST_HEADER_FIELDS_TOO_LARGE, 431);
    }

    public final function testUnavailableForLegalReasons(): void
    {
        $this->assertEquals(Error::UNAVAILABLE_FOR_LEGAL_REASONS, 451);
    }
}


tests/HttpSuccessTest.php
<?php

/*!
 * Copyright © 2020 Amin NAIRI
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

use PHPUnit\Framework\TestCase;
use Folded\Http\Success;

final class HttpSuccessTest extends TestCase
{
    public final function testOk(): void
    {
        $this->assertEquals(Success::OK, 200);
    }

    public final function testCreated(): void
    {
        $this->assertEquals(Success::CREATED, 201);
    }

    public final function testAccepted(): void
    {
        $this->assertEquals(Success::ACCEPTED, 202);
    }

    public final function testNonAuthoritativeInformation(): void
    {
        $this->assertEquals(Success::NON_AUTHORITATIVE_INFORMATION, 203);
    }

    public final function testNoContent(): void
    {
        $this->assertEquals(Success::NO_CONTENT, 204);
    }

    public final function testResetContent(): void
    {
        $this->assertEquals(Success::RESET_CONTENT, 205);
    }

    public final function testPartialContent(): void
    {
        $this->assertEquals(Success::PARTIAL_CONTENT, 206);
    }

    public final function testMultiStatus(): void
    {
        $this->assertEquals(Success::MULTI_STATUS, 207);
    }

    public final function testAlreadyReported(): void
    {
        $this->assertEquals(Success::ALREADY_REPORTED, 208);
    }

    public final function testInstanceManipulationsUsed(): void
    {
        $this->assertEquals(Success::IM_USED, 226);
    }
}


tests/HttpInformationTest.php
<?php

/*!
 * Copyright © 2020 Amin NAIRI
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

use PHPUnit\Framework\TestCase;
use Folded\Http\Information;

final class HttpInformationTest extends TestCase
{
    public final function testContinueRequest(): void
    {
        $this->assertEquals(Information::CONTINUE_REQUEST, 100);
    }

    public final function testSwitchingProtocols(): void
    {
        $this->assertEquals(Information::SWITCHING_PROTOCOLS, 101);
    }

    public final function testProcessing(): void
    {
        $this->assertEquals(Information::PROCESSING, 102);
    }

    public final function testEarlyHints(): void
    {
        $this->assertEquals(Information::EARLY_HINTS, 103);
    }
}


sources/Success.php
<?php

/**
 * Copyright 2020 Amin NAIRI
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
 * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

namespace Folded\Http;

final class Success
{
    public const ACCEPTED = 202;
    public const ALREADY_REPORTED = 208;
    public const CREATED = 201;
    public const IM_USED = 226;
    public const MULTI_STATUS = 207;
    public const NON_AUTHORITATIVE_INFORMATION = 203;
    public const NO_CONTENT = 204;
    public const OK = 200;
    public const PARTIAL_CONTENT = 206;
    public const RESET_CONTENT = 205;
}


sources/Information.php
<?php

/**
 * Copyright 2020 Amin NAIRI
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
 * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

namespace Folded\Http;

final class Information
{
    public const CONTINUE_REQUEST = 100;
    public const EARLY_HINTS = 103;
    public const PROCESSING = 102;
    public const SWITCHING_PROTOCOLS = 101;
}


sources/Redirection.php
<?php

/**
 * Copyright 2020 Amin NAIRI
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
 * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

namespace Folded\Http;

final class Redirection
{
    public const FOUND = 302;
    public const MOVED_PERMANENTLY = 301;
    public const MOVED_TEMPORARILY = 302;
    public const MULTIPLE_CHOICES = 300;
    public const NOT_MODIFIED = 304;
    public const PERMANENT_REDIRECT = 308;
    public const SEE_OTHER = 303;
    public const SWITCH_PROXY = 306;
    public const TEMPORARY_REDIRECT = 307;
    public const USE_PROXY = 305;
}


sources/ServerError.php
<?php

/**
 * Copyright 2020 Amin NAIRI
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
 * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

namespace Folded\Http;

final class ServerError
{
    public const BAD_GATEWAY = 502;
    public const GATEWAY_TIMEOUT = 504;
    public const HTTP_VERSION_NOT_SUPPORTED = 505;
    public const INSUFFICIENT_STORAGE = 507;
    public const INTERNAL_SERVER_ERROR = 500;
    public const LOOP_DETECTED = 508;
    public const NETWORK_AUTHENTICATION_REQUIRED = 511;
    public const NOT_EXTENDED = 510;
    public const NOT_IMPLEMENTED = 501;
    public const SERVICE_UNAVAILABLE = 503;
    public const VARIANT_ALSO_NEGOTIATES = 506;
}


sources/Error.php
<?php

/**
 * Copyright 2020 Amin NAIRI
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
 * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS IN THE SOFTWARE.
 */

declare(strict_types=1);

namespace Folded\Http;

final class Error
{
    public const BAD_REQUEST = 400;
    public const CONFLICT = 409;
    public const EXPECTATION_FAILED = 417;
    public const FAILED_DEPENDENCY = 424;
    public const FORBIDDEN = 403;
    public const GONE = 410;
    public const IM_A_TEAPOT = 418;
    public const LENGTH_REQUIRED = 411;
    public const LOCKED = 423;
    public const METHOD_NOT_ALLOWED = 405;
    public const MISDIRECTED_REQUEST = 421;
    public const NOT_ACCEPTABLE = 406;
    public const NOT_FOUND = 404;
    public const PAYLOAD_TOO_LARGE = 413;
    public const PAYMENT_REQUIRED = 402;
    public const PRECONDITION_FAILED = 412;
    public const PRECONDITION_REQUIRED = 428;
    public const PROXY_AUTHENTICATION_REQUIRED = 407;
    public const RANGE_NOT_SATISFIABLE = 416;
    public const REQUEST_HEADER_FIELDS_TOO_LARGE = 431;
    public const REQUEST_TIMEOUT = 408;
    public const TOO_EARLY = 425;
    public const TOO_MANY_REQUESTS = 429;
    public const UNAUTHORIZED = 401;
    public const UNAVAILABLE_FOR_LEGAL_REASONS = 451;
    public const UNPROCESSABLE_ENTITY = 422;
    public const UNSUPPORTED_MEDIA_TYPE = 415;
    public const UPGRADE_REQUIRED = 426;
    public const URI_TOO_LONG = 414;
}
